standards/branching.md

```diff
@@ -27,7 +27,9 @@ For example, if I'm (Corey) working on an bug fix task, I'll create a branch suc
 
 When work is completed and tested, make a Pull Request to the development branch. This PR requires review from another developer before it is merged.
 
-When code is ready to be locked in for demo, a Pull Request from `development` to `main` will be created and reviewed.
+When code is ready to be locked in for demo, a Pull Request from `development` to `staging` will be created and reviewed.
+
+When code is ready to be deployed, a Pull Request from `development` to `main` will be created and reviewed.
```

Comment:
should deployments go through `staging` and then to `main`?

Reply:
I believe the `staging` branch is centered around demos. 

Demos will often have the following:
1. Temporary pages filled with sample data.
2. Quick bug fixes that are not always in best practice (no testing, no long term data structures, etc..)
3. Hard set values that are not being served up from the server.

The above is designed to make the customer feel good about the application without running into dead ends and bugs within the code. One of the pros about the above is developers can keep working on development features that may get in the way of the demo without worrying about the `staging` branch. One of the cons on the approach is a small amount of overhead creating and maintaining the short lived staging branches for a demo. 

This is one of those blind spots that are often not talked about in the [atlassian workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) and classic [Gitflow](https://nvie.com/posts/a-successful-git-branching-model/) explanation so it is open for discussion. 

Why should deployments go though `staging` and then to `main`? What are the benefits and cons for doing so?